docs/source/tertiary/whole-genome-regression.rst

```diff
@@ -11,47 +11,65 @@ GloWGR: Whole Genome Regression
+Glow supports Whole Genome Regression (WGR) as GloWGR, a parallelized version of the `regenie <https://rgcgithub.github.io/regenie/>`_ method (see the `preprint <https://www.biorxiv.org/content/10.1101/2020.06.19.162354v2>`_). GloWGR supports two types of phenotypes:
+
+- Quantitative 
+- Binary
+
+Many steps of the GloWGR workflow explained in this page are common between the two cases. Any step that is different between the two has  separate explanations clearly marked by "for quantitative phenotypes" vs. "for binary phenotypes".
+
+-----------
+Performance
+-----------
+
+The following figures demonstrates the performance gain obtained by using parallelized GloWGR in comparision with single machine BOLT, fastGWA GRM, and regenie for quantitative and binary phenotypes.
+
+.. TODO: We need to add more information on the experiments behind this graphs.
 
 .. image:: ../_static/images/wgr_runtime.png
    :scale: 50 %
 
+.. TODO: Add a picture for binary
+
+--------
+Overview
+--------
+
 GloWGR consists of the following stages.
 
-- Block the genotype matrix across samples and variants.
-- Perform dimensionality reduction with ridge regression.
-- Estimate phenotypic values with ridge regression.
+- Block the genotype matrix across samples and variants
+- Perform dimensionality reduction with linear ridge regression
+- Estimate phenotypic predictors using
 
-.. note::
+  - **For quantitative phenotypes**: linear ridge regression
+  - **For binary phenotypes**: logistic ridge regression
+
+The following diagram provides an overview of the operations and data within the main stages of GlowWGR and their interrelationship.
+
+.. image:: ../_static/images/wgr_diagram.png
 
-   GloWGR currently supports only quantitative phenotypes.
 
 ----------------
 Data preparation
 ----------------
 
-GloWGR accepts three input datasources.
+GloWGR accepts three input data components.
 
-Genotype data
-=============
+.. _genotype-data:
 
-The genotype data may be read from any variant datasource supported by Glow, such as one read from
-:ref:`VCF, BGEN or PLINK <variant_data>`. For scalability, recommend ingesting flat genotype files into
-:ref:`Delta tables <vcf2delta>`.
+1. Genotype data
+================
 
-The DataFrame must also include a column ``values`` containing a numeric representation of each genotype. The genotypic
-values may not be missing, or equal for every sample in a variant (eg. all samples are homozygous reference).
+The genotype data may be read as a Spark DataFrame from any variant data source supported by Glow, such as :ref:`VCF, BGEN or PLINK <variant_data>`. For scalability and high-performance repeated use, we recommend storing flat genotype files into :ref:`Delta tables <vcf2delta>`.
+The DataFrame must include a column ``values`` containing a numeric representation of each genotype. The genotypic values may not be missing and should not be equal across all samples in a variant (e.g., all samples are homozygous reference).
```

Comment:
The "should not be equal" bit conflicts with the warning that we remove such rows. If we're removing them, users don't need to filter them out explicitly.

Reply:
removed this sentence.